### PR TITLE
Add validation for username in settings page

### LINKED ISSUE #2103: Improve the username input field of Create New User section(Settings Page)
**Summary**
Currently, the username field in the "Create New User" section can take any characters including spaces. Validation needs to be added in the username field so that it takes only alphanumeric characters and periods(.), similar to Gmail username validation.

**SCREENSHOT**
Screenshot from 2020-09-24 00-19-31

Reference can be taken from the above screenshot, a similar error message can be displayed if the format is wrong.

**ISSUE TYPE**
Code Refactor

**CONTACT PERSON**
[Saranya Jena](https://github.com/Saranya-jena)

Thanks for taking the time to read this issue. In order for Litmus to clarify your doubts effectively, provide us a few hours to reply back.

### PR DESCRIPTION
**Pull request by amaaniqbal**
Signed-off-by: Amaan Iqbal <amaaniqbal2786@gmail.com>

## Proposed changes
Add validation for `alphanumeric characters and period` to `username` field in the settings page.
Fixes #2103 

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- None

## Special notes for your reviewer:
 Enhancement PR

### CODE REVIEW

**Comment by amaaniqbal** [comment]
Sample Screenshots for the edit:
- Valid Username
correct

- Invalid Username
incorrect

@Saranya-jena @S-ayanide

**Comment by amaaniqbal** [comment]
@amityt @S-ayanide @Saranya-jena any updates?

**Comment by amaaniqbal** [comment]
@uditgaurav @arkajyotiMukherjee @ksatchit This PR is open since October. Can you please try reviewing it, and add the corresponding `Hacktoberfest` labels?

**Comment by amaaniqbal** [comment]
@S-ayanide @Saranya-jena Are these changes no longer required?

File: litmus-portal/frontend/src/utils/validate.ts [hunk 1]
```diff
@@ -19,6 +19,15 @@ export const validateEmail = (value: string) => {
   return false;
 };
 
+export const validateUsername = (value: string) => {
+  const usernameValid = /^[a-zA-Z0-9\.]+$/;
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/utils/validate.ts:23]: This regex does not allow _ @ and - in usernames, sure it excludes # but doesn't include those mentioned as well. You could please modify the regex!

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/utils/validate.ts (hunk 1 above)
Comment (amaaniqbal) [reply litmus-portal/frontend/src/utils/validate.ts]: @S-ayanide I updated the RegEx. Please have a look!

**Review by S-ayanide — APPROVED** [review]
Thanks for the quick update, LGTM

**Comment by amaaniqbal** [comment]
@Saranya-jena @amityt Can anyone of you try adding the second review?

**Comment by Saranya-jena** [comment]
Closing this PR as some of the users want to keep the username field simple and don't want validation for the same. Will reconsider this if required in the future.

**Comment by amaaniqbal** [comment]
@prithvi1307 Will this PR be eligible for Hacktoberfest tees?

**Pull request closed without merging**